src/relentless/optimize/objective.py

```diff
@@ -353,7 +357,12 @@ def compute(self, variables, directory=None):
         # run simulation and use result to compute gradient
         try:
             sim = self.simulation.run(self.potentials, directory)
-            sim_ens = sim[self.thermo]["ensemble"]
+            if isinstance(self.target, Ensemble) and isinstance(
+                self.thermo, EnsembleAverage
+            ):
+                sim_ens = sim[self.thermo]["ensemble"]
+            else:
+                sim_ens = sim[self.thermo]["_gsd"].filename
```

Comment:
This lets us very easily pull the current iterations gsd file for analysis! So this should be pretty straightforward to implement!